Core/Tests/Program.cs

```diff
@@ -21,124 +21,199 @@
 
 namespace Loyc.Tests
 {
+	/// <summary>
```

Comment:
This class is unnecessary - especially given its single-purpose nature. Instead you should use `IEnumerable<char>` or `IEnumerator<char>`:

~~~csharp
static IEnumerator<char> ConsoleChars()
{
	for (ConsoleKeyInfo k; (k = Console.ReadKey(true)).Key != ConsoleKey.Escape 
	                                              && k.Key != ConsoleKey.Enter;)
		yield return (int)k.KeyChar;
	yield return -1;
} 
~~~